docs/articles/documentation/recipes/basics/test-file-upload.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: docs
+title: Test File Upload
+permalink: /documentation/recipes/basics/test-file-upload.html
+---
+# Test File Upload
+
+This recipe shows how to test your application's file upload functionality with TestCafe.
+
+[Full Example Code](https://github.com/DevExpress/testcafe-examples/tree/master/examples/upload-files)
+
+Assume the following HTML:
+
+```html
+<body>
+    <form action="http://localhost:3000/upload" method="post" enctype="multipart/form-data">
+        <input id="upload-input" type="file" name="files" multiple />
+        <input type="submit" id="upload-btn" value="Upload Files"/>
+    </form>
+</body>
+```
+
+TestCafe's file upload emulation is differrent from what a real user would do to upload files.
+
+To manually upload files on this page, the user needs to perform the following steps:
+
+* Click the 'Choose File' button that opens the 'Choose file' dialog in the browser
+* Select the files
+* Begin the upload
+
+Since TestCafe cannot interact with the native dialog, it manipulates the `<input type=file>` elements' state with the `.setFilesToUpload` action.
+
+The sample page includes a `<form>` that contains:
+
+* an `<input>` element that holds the selected files
+* a `submit` button that sends the files to the server.
+
+Below is the full test code and a step-by-step explanation.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `./index.html`
+
+test('Select files to upload', async t => {
```

Comment:
`test('Check uploaded files',`

here and below.

Reply:
edited